### PR TITLE
Refactor `install.sh`

### PR DESCRIPTION
**Pull request by kasperski95**
I need to add more code to this file without making a mess.

Related: https://github.com/software-mansion/protostar/issues/605

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -1,96 +1,143 @@
...
-  Darwin)
-    PLATFORM="macOS"
-    ;;
-  *)
-    echo "unsupported platform: $PLATFORM"
-    exit 1
-    ;;
-esac
+PROTOSTAR_REPO="https://github.com/software-mansion/protostar"
+RESULT=""
```
Comment (mkaput) [inline install.sh:5]: 1. I have seen name `RETVAL` in the wild (for example here: https://github.com/rust-lang/rustup/blob/master/rustup-init.sh)
2. This statement is NOOP as far as I know bash. You can safely remove it.

File: install.sh [hunk 2]
```diff
@@ -1,96 +1,143 @@
...
+
+function create_protostar_directory() {
```
Comment (mkaput) [inline install.sh:7]:
```suggestion
create_protostar_directory() {
```

FYI The `function` keyword is unnecessary (really!) :)

File: install.sh [hunk 3]
```diff
@@ -1,96 +1,143 @@
...
+
+    if [[ ":$PATH:" != *":${_protostar_binary_dir}:"* ]]; then
+        echo >>$_profile && echo "export PATH=\"\$PATH:$_protostar_binary_dir\"" >>$_profile
+    fi
+    echo && echo "Detected your preferred shell is ${_pref_shell} and added Protostar to PATH. Run 'source ${_profile}' or start a new terminal session to use Protostar."
+    echo "Then, run 'protostar --help'."
+}
+
+function main() {
+    _provided_version_arg=$1
```
Comment (mkaput) [inline install.sh:100]: why are you prefixing names with `_`?

you can use `local` here

**Review by mkaput — CHANGES_REQUESTED** [review]

File: install.sh (hunk 3 above)
Comment (kasperski95) [reply install.sh:100]: To make locals stand out compared to function calls.

(I forgot about the `local` keyword in bash.)

File: install.sh (hunk 2 above)
Comment (kasperski95) [reply install.sh:7]: I know, but VS Code highlights `function` keyword differently. I'll remove it.

File: install.sh (hunk 3 above)
Comment (kasperski95) [reply install.sh]: Removed prefixes.

File: install.sh (hunk 2 above)
Comment (kasperski95) [reply install.sh]: Removed function keyword

File: install.sh (hunk 1 above)
Comment (kasperski95) [reply install.sh]: Removed global declaration. Renamed to retval

**Review by mkaput — APPROVED** [review]

**Pull request merged**